### PR TITLE
MWPW-136885: More SEO-friendly, improved accessibility in merch cards

### PR DESCRIPTION
**Pull request by Axelcureno**
Programmatically replaces heading tag names in cascading order to improve SEO and accessibility in Merch cards.
The hierarchal order is the following:
- 1st heading: H3
- 2nd heading: H4
- 3rd heading: H5

_Note: No merch card should have more than 3 headings._

Resolves: [MWPW-136885](https://jira.corp.adobe.com/browse/MWPW-136885)

**Test URLs:**
- Before: https://main--milo--axelcureno.hlx.page/docs/library/kitchen-sink/merch-card
- After: https://mwpw-136885--milo--axelcureno.hlx.page/docs/library/kitchen-sink/merch-card

### CODE REVIEW

**Review by mirafedas — APPROVED** [review]

File: libs/blocks/merch-card/merch-card.js [hunk 1]
```diff
@@ -125,30 +125,38 @@ const parseContent = (el, merchCard) => {
   ];
   let bodySlotName = 'body-xs';
   let headingMCount = 0;
+  let headingCount = 3;
```
Comment (npeltier) [inline libs/blocks/merch-card/merch-card.js:128]: can we call it headingSize?

Comment (npeltier) [reply libs/blocks/merch-card/merch-card.js:128]: count sounds weird to me

**Review by npeltier — APPROVED** [review]

File: libs/blocks/merch-card/merch-card.js [hunk 2]
```diff
@@ -125,30 +125,38 @@ const parseContent = (el, merchCard) => {
   ];
   let bodySlotName = 'body-xs';
```
[the remark this answers is not in the document]
Comment (Axelcureno) [reply libs/blocks/merch-card/merch-card.js]: Thx, addressed.

File: libs/blocks/merch-card/merch-card.js (hunk 1 above)
Comment (Axelcureno) [reply libs/blocks/merch-card/merch-card.js]: Sure, it's done.

File: libs/blocks/merch-card/merch-card.js [hunk 3]
```diff
@@ -125,30 +125,38 @@ const parseContent = (el, merchCard) => {
...
           }
           if (headingMCount === 2 && merchCard.variant === MINI_COMPARE_CHART) {
             slotName = 'heading-m-price';
           }
+          tagName = `H${headingCount}`;
+          headingCount += 1;
         }
         element.setAttribute('slot', slotName);
-        merchCard.append(element);
+        const newElement = createTag(tagName);
```
[the remark this answers is not in the document]
Comment (Axelcureno) [reply libs/blocks/merch-card/merch-card.js:151]: Fixed

**Comment by rgclayton** [comment]
A few observations. 

If I author the title with an `<h> tag` other than `<h3>` the font size is different. The DOM is changed to an `<h3>` though if it's the first title, like your code says.

To me it seems like if you are going to the trouble to change the title element to something specific the font size should stay consistent? Or maybe that's expected? Seems odd. 

If I author with an `<h1>`, `<h5>` or `<h6>` the title is not rendered at all. Why not look for these so they at least render?


[Sample page](https://mwpw-136885--milo--axelcureno.hlx.page/drafts/rclayton/test/merch-card-testing) I test with.


image

**Comment by Axelcureno** [comment]
> A few observations.
> 
> If I author the title with an `<h> tag` other than `<h3>` the font size is different. The DOM is changed to an `<h3>` though if it's the first title, like your code says.
> 
> To me it seems like if you are going to the trouble to change the title element to something specific the font size should stay consistent? Or maybe that's expected? Seems odd.
> 
> If I author with an `<h1>`, `<h5>` or `<h6>` the title is not rendered at all. Why not look for these so they at least render?
> 
> [Sample page](https://mwpw-136885--milo--axelcureno.hlx.page/drafts/rclayton/test/merch-card-testing) I test with.
> 
> image

@rgclayton thanks for the comment, you brought a good point. 
I started a discussion about it to make things easier for authors: https://github.com/orgs/adobecom/discussions/1799

File: libs/blocks/merch-card/merch-card.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (npeltier) [reply libs/blocks/merch-card/merch-card.js:151]: does not look like it

[the remark this answers is not in the document]
Comment (Axelcureno) [reply libs/blocks/merch-card/merch-card.js:151]: Sorry, I got confused with the another comment.
I tried doing your suggestion, but it seems element.attributes returns a `NamedNodeMap` which is not compatible with `createTag()`.
Screenshot 2024-01-26 at 2 36 25 PM

**Review by npeltier — CHANGES_REQUESTED** [review]
requesting discussion to be solved before going on :)

**Review by vladen — APPROVED** [review]

**Review by VKniaz — APPROVED** [review]

**Review by npeltier — APPROVED** [review]

**Pull request merged**